ega-data-api-res/src/main/java/eu/elixir/ega/ebi/reencryptionmvc/config/My2KCachePageFactory.java

```diff
@@ -81,22 +82,22 @@
     private final String awsSecretAccessKey;
     private final String awsEndpointUrl;
     private final String awsRegion;
-    private final String fireUrl;
-    private final String fireArchive;
-    private final String fireKey;
+    private final String fireURL;
+    private final String base64EncodedCredentials;
+        
+    @Autowired
+    private ArchiveAdapterService archiveAdapterService;
 
     public My2KCachePageFactory(Cache<String, EgaAESFileHeader> myHeaderCache,
```

Comment:
We need to refactor this class. We need to separate the access to S3 or Fire to their own services

Reply:
Tried to segregate the S3 and Fire code. Pls, review it again.